pkg/query-service/ee/usage/manager.go

```diff
@@ -0,0 +1,291 @@
+		repo:        &repo,
+		licenseRepo: licenseRepo,
+	}
+
+	if err := m.start(); err != nil {
+		return m, err
+	}
+
+	return m, nil
+}
+
+// start loads collects and exports any exported snapshot and starts the exporter
+func (lm *Manager) start() error {
+	// upload previous snapshots if any
+	err := lm.UploadUsage(context.Background())
+	if err != nil {
+		return err
+	}
+
+	// collect snapshot if incase it wasn't collect in (t - collectionFrequency)
+	err = lm.CollectCurrentUsage(context.Background())
+	if err != nil {
+		return err
+	}
+
+	go lm.UsageExporter(context.Background())
+
+	return nil
+}
+
+// CollectCurrentUsage checks if needs to collect usage data
+func (lm *Manager) CollectCurrentUsage(ctx context.Context) error {
+	// check the DB if anything exist  where timestamp > t - collectionFrequency
+	ts := time.Now().Add(-collectionFrequency)
+	alreadyCreated, err := lm.repo.CheckSnapshotGtCreatedAt(ctx, ts)
+	if err != nil {
+		return nil
+	}
+	if !alreadyCreated {
+		zap.S().Info("Collecting current usage")
+		exportError := lm.CollectAndStoreUsage(ctx)
+		if exportError != nil {
+			return exportError
+		}
+	} else {
+		zap.S().Info("Skipping current usage collection")
+	}
+	return nil
+}
+
+func (lm *Manager) UsageExporter(ctx context.Context) {
+	defer close(lm.terminated)
+
+	collectionTicker := time.NewTicker(collectionFrequency)
+	defer collectionTicker.Stop()
+
+	uploadTicker := time.NewTicker(uploadFrequency)
+	defer uploadTicker.Stop()
+
+	for {
+		select {
+		case <-lm.done:
+			return
+		case <-collectionTicker.C:
+			lm.CollectAndStoreUsage(ctx)
+		case <-uploadTicker.C:
+			lm.UploadUsage(ctx)
+		}
+	}
+}
+
+type TableSize struct {
+	Table             string `ch:"table"`
+	DiskName          string `ch:"disk_name"`
+	Rows              uint64 `ch:"rows"`
+	UncompressedBytes uint64 `ch:"uncompressed_bytes"`
+}
+
+func (lm *Manager) CollectAndStoreUsage(ctx context.Context) error {
+	snap, err := lm.GetUsageFromClickHouse(ctx)
+	if err != nil {
+		return err
+	}
+
+	license, err := lm.licenseRepo.GetActiveLicense(ctx)
+	if err != nil {
+		return err
+	}
+
+	activationId, _ := uuid.Parse(license.ActivationId)
+	// TODO (nitya) : Add installation ID in the payload
+	payload := model.UsagePayload{
+		UsageBase: model.UsageBase{
+			ActivationId:      activationId,
+			FailedSyncRequest: 0,
+		},
+		Metrics:      *snap,
+		SnapshotDate: time.Now(),
+	}
+
+	_, err = lm.repo.InsertSnapshot(ctx, payload)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (lm *Manager) GetUsageFromClickHouse(ctx context.Context) (*model.UsageSnapshot, error) {
+	tableSizes := []TableSize{}
+	snap := model.UsageSnapshot{}
+
+	// get usage from clickhouse
+	query := `
+		SELECT
+			table,
+			disk_name,
+			sum(rows) as rows,
+			sum(data_uncompressed_bytes) AS uncompressed_bytes
+		FROM system.parts
+		WHERE active AND (database in ('signoz_logs', 'signoz_metrics', 'signoz_traces')) AND (table in ('logs','samples_v2', 'signoz_index_v2'))
```

Comment:
Ah I see this is taking samples size only. Please rename the UsagePayload struct to reflect the same

Reply:
Have updated the struct